include/rcutils/env.h

```diff
@@ -58,6 +55,67 @@ RCUTILS_WARN_UNUSED
 bool
 rcutils_set_env(const char * env_name, const char * env_value);
 
+/// Retrieve the value of the given environment variable if it exists, or "".
+/** The c-string which is returned in the env_value output parameter is only
+ * valid until the next time this function is called, because it is a direct
+ * pointer to the static storage.
+ * The variable env_value populated by this function should never have free()
+ * called on it.
+ * If the environment variable is not set, an empty string will be returned.
+ *
+ * In both cases, environment variable set or unset, NULL is returned unless
+ * an exception has occurred, in which case the error string is returned.
+ * For example:
+ *
+ * ```c
+ * #include <stdio.h>
+ * #include <rcutils/env.h>
+ * const char * env_value;
+ * const char * error_str;
+ * error_str = rcutils_get_env("SOME_ENV_VAR", &env_value);
+ * if (error_str != NULL) {
+ *   fprintf(stderr, "Error getting env var: %s\n", error_str);
+ * }
+ * printf("Valued of 'SOME_ENV_VAR': %s\n", env_value);
+ * ```
+ *
+ * This function cannot be concurrently called together with rcutils_set_env (or any platform specific equivalent)
+ * on different threads, but multiple concurrent calls to this function are thread safe.
+ *
+ * \param[in] env_name the name of the environment variable
+ * \param[out] env_value pointer to the value cstring, or "" if unset
+ * \return NULL on success (success can be returning an empty string), or
+ * \return an error string on failure.
+ */
+RCUTILS_PUBLIC
+RCUTILS_WARN_UNUSED
+const char *
+rcutils_get_env(const char * env_name, const char ** env_value);
+
+/// Retrieve the full path to the home directory.
+/**
+ * The c-string which is returned is only valid until the next time this
+ * function is called, because it is a direct pointer to the static storage.
+ * Also note that the string returned here should *not* be freed.
+ *
+ * The function first tries to get the HOME environment variable.
+ * If that variable exists and is non-empty, that will be returned.
+ * Otherwise, the function tries to get the USERPROFILE environment variable.
+ * If that variable exists and is non-empty, that will be returned.
+ * If neither exists, NULL will be returned.
+ * The above algorithm is portable across both Unix and Windows.
```

Comment:
I know that you are just moving this, but this documentation is actually incorrect.  On Unix, `rcutils_get_home_dir` will try to get the HOME environment variable only.  If that fails, it returns NULL.  On Windows, `rcutils_get_home_dir` will try to get the HOME environment variable.  If that fails, it tries to get the USERPROFILE environment variable.  If that fails, it returns NULL.

Reply:
Fixed in 7eb69e04cecd44bd3bf0a47c38bae5b05a873ba3